src/Features/Core/Portable/Completion/Providers/SymbolCompletionItem.cs

```diff
@@ -12,94 +14,278 @@
             SupportedPlatformData supportedPlatforms = null,
+            bool isArgumentName = false,
+            ImmutableDictionary<string, string> properties = null,
             CompletionItemRules rules = null)
-        : base(completionProvider, displayText, filterSpan,
-           descriptionFactory: null, glyph: symbols[0].GetGlyph(),
-           sortText: symbols[0].Name, filterText: filterText, preselect: preselect, showsWarningIcon: supportedPlatforms != null, rules: rules)
         {
-            this.InsertionText = insertionText;
-            this.Position = position;
-            this.Symbols = symbols;
-            this.Context = context;
-            _supportedPlatforms = supportedPlatforms;
+            return Create(
+                completionProvider,
+                displayText: displayText,
+                filterSpan: filterSpan,
+                symbols: ImmutableArray.Create(symbol),
+                contextPosition: contextPosition,
+                descriptionPosition: descriptionPosition,
+                sortText: sortText,
+                insertionText: insertionText,
+                glyph: glyph,
+                filterText: filterText,
+                preselect: preselect,
+                supportedPlatforms: supportedPlatforms,
+                isArgumentName: isArgumentName,
+                properties: properties,
+                rules: rules);
         }
 
-        public SymbolCompletionItem(
-            CompletionListProvider completionProvider,
-            string displayText,
-            string insertionText,
-            string filterText,
-            TextSpan filterSpan,
-            int position,
-            List<ISymbol> symbols,
-            string sortText,
-            AbstractSyntaxContext context,
-            Glyph glyph,
-            bool preselect = false,
-            SupportedPlatformData supportedPlatforms = null,
-            CompletionItemRules rules = null)
-        : base(completionProvider, displayText, filterSpan,
-           descriptionFactory: null, glyph: glyph,
-           sortText: sortText, filterText: filterText, preselect: preselect, showsWarningIcon: supportedPlatforms != null, rules: rules)
+        public static string EncodeSymbols(IReadOnlyList<ISymbol> symbols)
         {
-            this.InsertionText = insertionText;
-            this.Position = position;
-            this.Symbols = symbols;
-            this.Context = context;
-            _supportedPlatforms = supportedPlatforms;
+            if (symbols.Count > 1)
+            {
+                return string.Join("|", symbols.Select(s => EncodeSymbol(s)));
+            }
+            else if (symbols.Count == 1)
+            {
+                return EncodeSymbol(symbols[0]);
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        public static string EncodeSymbol(ISymbol symbol)
+        {
+            return SymbolId.CreateId(symbol);
```

Comment:
I feel like this should be SymbolKey, just with well defined tostring/fromstring methods on it.